dali/base/dadfs.cpp

```diff
@@ -118,6 +118,16 @@ inline unsigned groupDistance(IGroup *grp1,IGroup *grp2)
     return grp1->distance(grp2);
 }
 
+inline StringBuffer &appendEnsurePathSepChar(StringBuffer &dest, StringBuffer &newPart, char psc)
+{
+    addPathSepChar(dest, psc);
+    if (isPathSepChar(newPart.charAt(0)))
```

Comment:
Is here any potential for newPart to be zero length?  (e.g. was '/' before trailing '/' removed?  Is so the following could go wrong.  (The test would pass, but newPart.str() would return a pointer passed the end of string.

Probably better is to pass newPart.str() as a const char * and then check *part, or also check the length.